department-of-veterans-affairs/vets-website
# Supplemental Claims | Implement feature flag

## Original Ticket

[#48165](https://github.com/department-of-veterans-affairs/va.gov-team/issues/48165)

## URL of change

`/decision-reviews/supplemental-claim/request-supplemental-claim-form-20-0995/`

## Background

A `supplemental_claim` feature flag has been added to allow for a staged rollout of the Supplemental Claim form. This PR renders a work-in-progress alert, which shouldn't ever been seen unless the user knows the URL to the form in production. It does not currently exist in production, but during rollout, the feature flag WIP alert will prevent users from accessing the page.

<details><summary>WIP alert</summary>


<img width="782" alt="work-in-progress message with, we're still working on this feature in the header and we're rolling out supplemental claims form (20-0995) in staging. It's not quite ready yet. Please check back again soon. Followed by a link back to the supplemental claims information page" src="https://user-images.githubusercontent.com/136959/198734022-264fe0c6-1d41-4a93-a33c-47a98a4d6cf3.png"></details>

## Steps to test

1. Pull this branch in locally or test in a review instance
2. If you don't have React dev tools installed, you may have to wait to test this in staging
3. Log in to any user (usually use 228)
4. Go to the URL of change
5. Open the React dev tools and open the Components tab
6. Find and select the innermost `App` component
7. Toggle the `show995` property to see the WIP alert

1. Go to URL of change plus `/start` to test sub-task
2. Repeat steps 5-7 to toggle the `show995` property

## Code changes

- Updated Intro page content per design feedback
- Add WIP alert
- Update SubTask container to show WIP alert based on feature toggle
- Updated unit tests
- Update e2e test

## How was your code tested

Unit & e2e test (only passing for app/intro pages. Still a WIP; disabled in CI)

## Acceptance criteria
- [x] Add WIP alert to App based on feature flag setting
- [x] Update e2e test
- [x] All tests passing (sorta)

## Definition of done
